### PR TITLE
TYP: some types for pandas/core/arrays/sparse/dtype.py

### PR DESCRIPTION
**Pull request by simonjayhawkins**
broken off #28339

### CODE REVIEW

File: pandas/core/arrays/sparse/dtype.py [hunk 1]
```diff
@@ -223,7 +223,7 @@ def construct_from_string(cls, string):
             raise TypeError(msg)
 
     @staticmethod
-    def _parse_subtype(dtype):
+    def _parse_subtype(dtype) -> Tuple[str, Union[str, bool]]:
```
Comment (jbrockmendel) [inline pandas/core/arrays/sparse/dtype.py]: not necessarily for this PR, but the `Union[str, bool]` is kind of weird.  maybe split into separate return values with clearer types?

Comment (jbrockmendel) [reply pandas/core/arrays/sparse/dtype.py]: can `dtype` be typed as `str`?

**Comment by jbrockmendel** [comment]
Pending typing `dtype` as str (or a reason why we cant/shouldnt), LGTM

File: pandas/core/arrays/sparse/dtype.py (hunk 1 above)
Comment (simonjayhawkins) [reply pandas/core/arrays/sparse/dtype.py]: maybe we could cast has_fill_value to bool before returning. cast as in the dynamic bool(...) sense not cast from typing?

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Thanks !

**Pull request merged**